### ISSUE TITLE
Enable users to encrypt data at rest

### ISSUE THREAD

**Issue by michmike** [goharbor/harbor#11088]
* Enable project admins to decide whether to encrypt data at rest
* Enable project admins to bring their own certificate for encryption

this would benefit service providers and internal IT for sensitive images. This is a compliance requirement for certain large organizations

**Comment by reasonerjt** [comment]
I think this has to be done at the infrastructure level, for example encrypt the filesystem or storage service.

**Comment by tianon** [comment]
I agree, I don't think there's a reasonable way that Harbor could transparently handle encrypting the images without it being only a minor speed bump to retrieve whatever key Harbor used in order to decrypt them -- projects like https://github.com/containerd/imgcrypt which seek to encrypt images through the entire pipeline and decrypt them in the runtime seem like a much better fit for this problem.

**Comment by tianon** [comment]
(Another solution for admins who only need "at rest" encryption is to use an encrypted filesystem for storage, which would be completely transparent to Harbor.)
